### PR TITLE
[IPv6] Add Unit Test for FreeRTOS_UDP_IP & FreeRTOS_UDP_IPv4

### PR DESCRIPTION
**Pull request by ActoryOu**
[IPv6] Add Unit Test for FreeRTOS_UDP_IP & FreeRTOS_UDP_IPv4.

Description
-----------

Separate FreeRTOS_UDP_IP unit test cases into FreeRTOS_UDP_IP/FreeRTOS_UDP_IPv4 and add test cases to reach 100% coverage.

Test Steps
-----------

Unit Test
```
cmake -S test/unit-test -B test/unit-test/build/ -DCMAKE_BUILD_TYPE=Debug
make -C test/unit-test/build/ all
```
Coverage Test
```
make -C ${BUILD_DIR} coverage
lcov --list --rc lcov_branch_coverage=1 ${BUILD_DIR}coverage.info
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_UDP_IP.c [hunk 1]
```diff
@@ -96,28 +96,31 @@ void vProcessGeneratedUDPPacket( NetworkBufferDescriptor_t * const pxNetworkBuff
...
+        /* Map the UDP packet onto the start of the frame. */
+
+        /* MISRA Ref 11.3.1 [Misaligned access] */
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
+        pxUDPPacket = ( ( UDPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
+
+        switch( pxUDPPacket->xEthernetHeader.usFrameType )
+        {
+            #if ( ipconfigUSE_IPv4 != 0 )
```
Comment (shubnil) [inline source/FreeRTOS_UDP_IP.c:110]: Do we need v4 and v6 flags in unit tests. I think we decided that both flags will always be enabled for unit test.

Comment (ActoryOu) [reply source/FreeRTOS_UDP_IP.c:110]: This is original source code. We need that flag to differentiate IPv4/IPv6.

File: test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c [hunk 2]
```diff
@@ -74,44 +72,25 @@
...
-    xInterface->pxEndPoint = xEndPoint;
-    xInterface->pxNext = NULL;
+    vProcessGeneratedUDPPacket( NULL );
 }
 
-/*
- * @brief Test what happens if the packet cannot be sent due to
- *        the address not being resolved.
+/**
+ * @brief Pass IPv6 packet to IPv4 API to handle.
```
Comment (kar-rahul-aws) [inline test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c:91]: Comment should be : Pass IPv4 packet to IPv4 API to handle

File: test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c [hunk 3]
```diff
@@ -124,998 +103,149 @@ void test_vProcessGeneratedUDPPacket_CantSendPacket( void )
...
-    TEST_ASSERT_EQUAL( pdFAIL, xResult );
-    TEST_ASSERT_EQUAL( pdFALSE, xIsWaitingARPResolution );
+    catch_assert( xProcessReceivedUDPPacket( &xNetworkBuffer, usPort, &xIsWaitingForARPResolution ) );
 }
 
-/*
- * @brief Test when there is no listening socket but the packet seems like
- *        a late DNS response.
+/**
+ * @brief Pass IPv6 packet to IPv4 API to handle.
```
Comment (kar-rahul-aws) [inline test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c:178]: Comment should be : Pass IPv4 packet to IPv4 API to handle.

Comment (ActoryOu) [reply test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c]: Updated

File: test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c (hunk 2 above)
Comment (ActoryOu) [reply test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c]: Updated

**Review by shubnil — APPROVED** [review]

**Pull request merged**